sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/InMemoryTableScanExec.scala

```diff
@@ -166,4 +170,32 @@ case class InMemoryTableScanExec(
   protected override def doExecuteColumnar(): RDD[ColumnarBatch] = {
     columnarInputRDD
   }
+
+  // ==================================================================
+  // Methods for AQE
+  // ==================================================================
+
+  @transient
+  private lazy val future: FutureAction[Unit] = {
+    val rdd = cachedPlan.execute()
+    sparkContext.submitJob(
+      rdd,
+      (_: Iterator[InternalRow]) => (),
+      (0 until rdd.getNumPartitions).toSeq,
+      (_: Int, _: Unit) => (),
+      ()
+    )
+  }
+
+  override def isMaterialized: Boolean = relation.isMaterialized || super.isMaterialized
```

Comment:
why do we consider `relation.isMaterialized`?

Reply:
If the `InMemoryTableScanExec` is materialized, we should not run a new job again. Add the check is to avoid AQE framework call `doMaterialize`.

Reply:
it seems `relation.isMaterialized` is not accuate enough, removed it